api/errors.go

```diff
@@ -27,7 +27,13 @@ var (
 
 type ErrStorageError struct{ Err error }
 
-func (e ErrStorageError) Error() string { return fmt.Sprintf("storage error: %s", e.Err.Error()) }
+func (e ErrStorageError) Error() string {
+	if e.Err != nil {
+		return fmt.Sprintf("storage error: %s", e.Err.Error())
+	}
+	// ErrStorageError shouldn't be constructed with a nil Err, but format it just in case.
+	return "generic storage error"
```

Comment:
do we have any callsites that can intentionally pass nil? I'm not aware of any place where nil indicates an error. which if there isn't, then this message should say something about the error being mishandled, rather than there being a "generic" error

Reply:
:+1: reworded